### ISSUE TITLE
"Styles" submenu no longer display all available styles by default for a given layer

### ISSUE THREAD

**Issue by jfbourdon** [qgis/QGIS#57251]
### What is the bug or the crash?

All available styles for a given layer are no longer displayed by default in the "Styles" submenu. This feature was introduced in 3.30 (#50812) to fix #37434.

Style submenu in QGIS 3.30.2 and 3.32.2:
![3 30 2](https://github.com/qgis/QGIS/assets/32580398/88e61855-a948-4f4f-b2df-479c69d996a0)

Style submenu in 3.34.4 and 3.36.0:
![3 34 4](https://github.com/qgis/QGIS/assets/32580398/344ca754-f965-4b7d-84d6-d05a3aaedf90)

### Steps to reproduce the issue

1. Create a GPKG containing a layer with two associated styles
```python
# Creation of the base layer
lyr = QgsVectorLayer("LineString?crs=EPSG:3857", "line", "memory")
wkt = 'LineString (0 0, 10 0, 5 5)'
geom = QgsGeometry.fromWkt(wkt)
fet = QgsFeature()
fet.setGeometry(geom)
lyr.dataProvider().addFeature(fet)

# Write layer as GeoPackage
path_gpkg = "C:/Temp/line.gpkg"
save_options = QgsVectorFileWriter.SaveVectorOptions()
QgsVectorFileWriter.writeAsVectorFormatV3(lyr, path_gpkg, QgsCoordinateTransformContext(), save_options)

# Store two different styles inside the GeoPackage
lyr = QgsVectorLayer(path_gpkg)
for color in ["red","blue"]:
    symbol = QgsLineSymbol.createSimple({'line_style': 'dash', 'color': color})
    lyr.renderer().setSymbol(symbol)
    lyr.saveStyleToDatabase(name=f"dash_{color}", description="", useAsDefault=False, uiFileContent="")
```
2. Load in QGIS the resulting GeoPackage and check if the styles are directly available in the "Styles" submenu of the layer in the Layers Panel.

### Versions

QGIS 3.34.4
QGIS 3.36.0

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
@elpaso, you may be interested in this bug report.

**Comment by elpaso** [comment]
> This feature was introduced in 3.30 (#50812) to fix #37434.

Actually, that is not correct.

The root cause of the issue has been there for years, it has probably surfaced after my changes but here is a test with commit 1373d2bcfcc Merge pull request #50812 from elpaso/bugfix-gh37434-default-style-name, as you can see it works as expected:

![immagine](https://github.com/qgis/QGIS/assets/142164/3a42de40-5de4-44aa-be80-81ab6932ff0e)

**Comment by agiudiceandrea** [comment]
@elapso, since your PR has been merged in 3.30.0, while the issue didn't occur using QGIS 3.30.2 or 3.32.2, it looks like some other PR may have been introduced the issue after your PR.
